### PR TITLE
add GitHub action to trigger Docker Desktop e2e tests with Compose edge version

### PR DESCRIPTION
**Pull request by glours**
**What I did**
Add a GitHub action job to trigger Docker Desktop e2e tests with the latest edge version of Compose to early detect integration issue.

**Related issue**
https://docker.atlassian.net/browse/ENV-120

**(not mandatory) A picture of a cute animal, if possible in relation to what you did**
image

### CODE REVIEW

File: .github/workflows/merge.yml [hunk 1]
```diff
@@ -118,3 +120,23 @@ jobs:
...
+
+  desktop-edge-test:
+    runs-on: ubuntu-latest
+    needs: bin-image
+    steps:
+      -
+        name: Trigger Docker Desktop e2e with edge version
+        uses: actions/github-script@v6
+        with:
+          github-token: ${{ secrets.DOCKERPUBLICBOT_WRITE_PAT }}
```
Comment (mat007) [inline .github/workflows/merge.yml:132]: That’s a token for Docker Hub, isn’t it? I’m not sure it’s going to help there.

Comment (glours) [reply .github/workflows/merge.yml:132]: Yep, that's why this PR is still a draft, I also need to update the way I'll get the image tag to pass to the workflow

**Review by mat007 — APPROVED** [review]
Makes sense!

**Review by ndeloof — APPROVED** [review]

**Review by milas — APPROVED** [review]
🎉
